app/main.py

```diff
@@ -1 +1,40 @@
-# write your code here
+from __future__ import annotations
+
+
+class Animal:
+    alive = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.health = health
+        self.name = name
+        self.hidden = hidden
+
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (
+            f"{{"
+            f"Name: {self.name}, "
+            f"Health: {self.health}, "
+            f"Hidden: {self.hidden}}}"
+        )
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    @classmethod
+    def bite(cls, other_animal: Herbivore) -> None:
+        if (other_animal.hidden is False
+                and isinstance(other_animal, Herbivore)):
+            other_animal.health -= 50
+        if other_animal.health <= 0:
```

Comment:
Your check for the health of animal should be on nested into first `if` statement